Updated engine.mdx and assertions.mdx to link to each other. (#4000)
langchain-ai/docs: src/langsmith/assertions.mdx

- You want to capture acceptance criteria in plain English without leaving the review flow.

<Note>

> Assertions are available on [single-run annotation queues](/langsmith/annotation-queues#single-run-annotation-queues). [Pairwise queues](/langsmith/annotation-queues#pairwise-annotation-queues) are unchanged. Assertions are currently UI-only. The [SDK guide](/langsmith/annotation-queues-sdk) covers programmatic access to feedback configs and queue rubrics, not assertions.

Reviewer: ```suggestion
Assertions are available on [single-run annotation queues](/langsmith/annotation-queues#single-run-annotation-queues). [Pairwise queues](/langsmith/annotation-queues#pairwise-annotation-queues) are unchanged. Assertions are available in the LangSmith UI only.
```

Seems a little weird to link to the SDK docs if assertions aren't supported there yet.

Author: updated as suggested

</Note>

<Tip>

> The [LangSmith Engine](/langsmith/engine#add-offline-examples) can auto-propose assertions for production traces flagged as recurring issues. Open an issue's offline examples flow to review, edit, or extend the Engine's proposed assertions before saving them to a dataset.

Reviewer: ```suggestion
[LangSmith Engine](/langsmith/engine#add-offline-examples) can auto-propose assertions for production traces flagged as recurring issues. Open an issue's offline examples flow to review, edit, or extend the Engine's proposed assertions before saving them to a dataset.
```

I assume we should refer to products without the `the`

Author: removed `The`
